### PR TITLE
Add proto.

### PR DESCRIPTION
**Pull request by siarheishut**
Include protobuf.
Add proto files for futures commits.

### CODE REVIEW

File: proto/transport_catalog.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  repeated int32 stops = 6;
+  repeated int32 endpoints = 7;
+}
+
+message Coords {
+  double latitude = 1;
+  double longitude = 2;
+}
+
+message StopInfo {
```
Comment (foxfrikses) [inline proto/transport_catalog.proto:23]: Simply `Stop` and `Bus`, without `Info` suffix.

File: proto/transport_catalog.proto [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+message Database {
+  repeated BusInfo buses = 1;
+  repeated StopInfo stops = 2;
+}
+
+message TransportCatalog {
+  Database database = 1;
+  RouteManager.RouteManager route_manager = 2;
+  RenderingSettings rendering_settings = 3;
+}
```
Comment (foxfrikses) [inline proto/transport_catalog.proto:38]: Maybe you then separate this file into two: `transport_catalog.proto` and `database.proto`

File: CMakeLists.txt [hunk 3]
```diff
@@ -12,6 +12,44 @@ FetchContent_Declare(
...
+
+FetchContent_GetProperties(Protobuf SOURCE_DIR Protobuf_SOURCE_DIR)
+include(${Protobuf_SOURCE_DIR}/cmake/protobuf-generate.cmake)
+
+include_directories(
+        ${Protobuf_SOURCE_DIR}/src
+        ${Protobuf_SOURCE_DIR}/third_party/abseil-cpp
+)
+
+add_library(my_proto
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply CMakeLists.txt:32]: Done.

File: proto/transport_catalog.proto [hunk 4]
```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+import "route_manager.proto";
+import "settings.proto";
+
+package TransportCatalog;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/transport_catalog.proto:6]: Done.

File: proto/settings.proto [hunk 5]
```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+message RoutingSettings {
+  int32 bus_wait_time = 1;
+  double bus_velocity = 2;
+}
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/settings.proto:6]: Done.

File: proto/settings.proto [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  kStopLabels = 3;
+};
+
+message Frame {
+  double width = 1;
+  double height = 2;
+  double padding = 3;
+}
+
+message Point {
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/settings.proto:21]: Done.

File: proto/settings.proto [hunk 7]
```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+message RoutingSettings {
+  int32 bus_wait_time = 1;
+  double bus_velocity = 2;
+}
+
+enum MapLayer {
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/settings.proto:8]: Done.

File: proto/transport_catalog.proto [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+message Coords {
+  double latitude = 1;
+  double longitude = 2;
+}
+
+message StopInfo {
+  string stop_name = 1;
+  Coords coords = 2;
+  repeated int32 buses_id = 4;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/transport_catalog.proto:26]: Done.

File: proto/transport_catalog.proto [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+package TransportCatalog;
+
+message BusInfo {
+  string bus_name = 1;
+  int32 stop_count = 2;
+  int32 unique_stop_count = 3;
+  int32 length = 4;
+  double curvature = 5;
+  repeated int32 stops = 6;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/transport_catalog.proto:14]: Done.

File: proto/transport_catalog.proto [hunk 10]
```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+import "route_manager.proto";
+import "settings.proto";
+
+package TransportCatalog;
+
+message BusInfo {
+  string bus_name = 1;
+  int32 stop_count = 2;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/transport_catalog.proto:10]: Done.

File: proto/transport_catalog.proto [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  int32 unique_stop_count = 3;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/transport_catalog.proto:11]: Done.

File: proto/transport_catalog.proto (hunk 1 above)
Comment (siarheishut) [reply proto/transport_catalog.proto:23]: Done.

File: proto/transport_catalog.proto (hunk 2 above)
Comment (siarheishut) [reply proto/transport_catalog.proto:38]: Done.

File: proto/route_manager.proto [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+message Route {
+  uint32 from = 1;
+  uint32 to = 2;
+  uint32 last_edge_id = 3;
+}
+
+message RouteManager {
+  repeated Edge edges = 1;
+  repeated Route routes = 2;
+  RoutingSettings routing_settings = 3;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/route_manager.proto:25]: Done.

File: proto/route_manager.proto [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  uint32 start_idx = 6;
+}
+
+message Route {
+  uint32 from = 1;
+  uint32 to = 2;
+  uint32 last_edge_id = 3;
+}
+
+message RouteManager {
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/route_manager.proto:22]: Done.

File: proto/route_manager.proto [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  uint32 bus_id = 3;
+  double weight = 4;
+  uint32 span_count = 5;
+  uint32 start_idx = 6;
+}
+
+message Route {
+  uint32 from = 1;
+  uint32 to = 2;
+  uint32 last_edge_id = 3;
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply proto/route_manager.proto]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**